File: studentprogressboard/studentprogressboard/Model/StudentCourse.cs
using studentprogressboard.Model.Base;
using studentprogressboard.Model.Enums;
using studentprogressboard.Model.Interfaces;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace studentprogressboard.Model
{
    public class StudentCourse : Entity, IEntity
    {
        public int? Grade { get; set; }
        public int SemesterId { get; set; }
        public int CourseId { get; set; }
        public CourseStatus? CourseStatus { get; set; } 


        [ForeignKey(nameof(SemesterId))]
        [NotMapped]
        public virtual Semester? Semester { get; set; }


        [ForeignKey(nameof(CourseId))]
        [NotMapped]
        public virtual Course? Course { get; set; }
    }
}

File: studentprogressboard/studentprogressboard.Client/src/app/_models/student-course.ts
import { CourseStatus } from "../_enums/course-status";
import { CourseType } from "../_enums/course-type";
import { RequirementType } from "../_enums/requirement-type";

export class StudentCourse{
    id: number;
    name: string;
    credits: number;
    grade: number;
    courseStatus: CourseStatus;
    courseType: CourseType;
    requirementType: RequirementType;
    semesterId: number;
    
    courseId: number;
}